docs: add Tasklist REST API docs [to master]

ce-dmelnych opened the pull request:
## What is the purpose of the change
PR with the same changes as here: https://github.com/camunda/camunda-platform-docs/pull/1870
but it points to `main` branch instead of [docs-apis-clients-tab](https://github.com/camunda/camunda-platform-docs/tree/docs-apis-clients-tab) branch.

Adding documentation for developed REST API in addition to existing GraphQL API for Tasklist service in the scope of:
https://github.com/camunda/product-hub/issues/908 epic and https://github.com/camunda/tasklist/issues/2546 implementation task.

_Briefly describe the change you are making, this helps the reviewer by providing an overview._

## Are there related marketing activities

_Are there any marketing activities related to the related feature or component? If so, please briefly describe them._

## When should this change go live?

_Does this change need to be released before a certain date or milestone? If so, when?_

## PR Checklist

- [ ] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [ ] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

akeller commented:
The biggest piece I'm missing here is any acknowledgment that we have two APIs for Tasklist. We should add this. Is there any missing functionality between the two? Something you can't do in one that you can do in another?

Are we ready to add a note to the GraphQL docs that this will be phased out?

ce-dmelnych commented on docs/components/tasklist/introduction-to-tasklist.md:
```diff
@@ -10,6 +10,12 @@ With Tasklist, orchestrate human workflows critical to your business and reduce
+and a [REST API](/docs/apis-clients/tasklist-api-rest/tasklist-api-rest-overview.md). Both APIs provide equal capabilities,
+allowing you to build your own applications or use the general [UI](/docs/components/tasklist/userguide/using-tasklist.md) that we have prepared for you.
+
+Please note that while both GraphQL and REST APIs are currently available, but the GraphQL API will be deprecated in future releases,
```
@marcosgvieira please pay attention to this note regarding the plans for GraphQL deprecation/future updates.

marcosgvieira commented:
> The biggest piece I'm missing here is any acknowledgment that we have two APIs for Tasklist. We should add this. Is there any missing functionality between the two? Something you can't do in one that you can do in another?
> 
> Are we ready to add a note to the GraphQL docs that this will be phased out?

@akeller, I added a note to the current GraphQL API and also in the tasklist overview - we are still not adding a deadline for the GraphQL but communicating that we plan to deprecate it in future releases.

vsgoulart commented on docs/apis-clients/tasklist-api-rest/controllers/tasklist-api-rest-task-controller.md:
```diff
@@ -0,0 +1,328 @@
+
+All request body parameters are optional.
+
+> NOTE: Only one of [searchAfter, searchAfterOrEqual, searchBefore, searchBeforeOrEqual] search options must be present in request.
```
```suggestion
> NOTE: Only one of [searchAfter, searchAfterOrEqual, searchBefore, searchBeforeOrEqual] search options must be present at once in a request.
```

ce-dmelnych replied:
corrected

vsgoulart commented on docs/apis-clients/tasklist-api-rest/schemas/models/date-filter-input.mdx:
````diff
@@ -0,0 +1,88 @@
+
+```json
+{
+  "from": "string",
+  "to": "string"
````
If it's a request example I believe the values should be an ISO string date as mentioned bellow, not just any string

ce-dmelnych replied:
added examples

vsgoulart commented on docs/apis-clients/tasklist-api-rest/schemas/models/task-order-by.mdx:
````diff
@@ -0,0 +1,88 @@
+
+```json
+{
+  "field": "string",
+  "order": "string"
````
Same comment from above about the ISO dates

ce-dmelnych replied:
added examples

vsgoulart commented on docs/apis-clients/tasklist-api-rest/schemas/responses/variable-response.mdx:
```diff
@@ -0,0 +1,107 @@
+
+#### [<code style={{ fontWeight: 'normal' }}>VariableResponse.<b>previewValue</b></code>](#)<Bullet />`string` <Badge class="secondary" text="non-null"/>
+
+> value preview (limited in size)
```
I think we should mention the size limit of this preview value

vsgoulart commented on docs/apis-clients/tasklist-api-rest/schemas/responses/variable-response.mdx:
```diff
@@ -0,0 +1,107 @@
+
+#### [<code style={{ fontWeight: 'normal' }}>VariableResponse.<b>id</b></code>](#)<Bullet />`string` <Badge class="secondary" text="non-null"/>
+
+> id of the variables
```
```suggestion
> id of the variable
```

ce-dmelnych replied:
thanks)
fixed typo

vsgoulart commented on docs/apis-clients/tasklist-api-rest/schemas/responses/task-response.mdx:
```diff
@@ -0,0 +1,169 @@
+
+#### [<code style={{ fontWeight: 'normal' }}>TaskSearchResponse.<b>taskDefinitionId</b></code>](#)<Bullet />`string` <Badge class="secondary" text="non-null"/>
+
+> Task Definition ID (node BPMN id) of the process
```
```suggestion
> User Task ID from the BPMN definition
```

marcosgvieira commented on docs/apis-clients/tasklist-api/tasklist-api-graphql-to-rest-migration.md:
```diff
@@ -8,10 +8,9 @@ description: "This article provides a guide for developers to migrate from Graph
-As the popularity of GraphQL continues to rise as an alternative to REST APIs,
-many developers have implemented it in their projects. However, this can often lead to challenges,
+As the popularity of GraphQL continues to rise as an alternative to REST APIs, many developers have implemented it in their projects. However, this can often lead to challenges,
 such as using multiple API protocols and maintaining compatibility with existing systems.
```
```suggestion
We want to provide you with the information you need to successfully migrate from our GraphQL API to our new REST API version. In this document, we'll explain the differences between the two APIs and provide guidance on how to make the switch.

GraphQL has been a popular and valuable tool for many of our customers, but we recognize that there are certain advantages to using a RESTful architecture. Our new REST API version provides a more structured and predictable way of accessing our data, which should lead to improved performance and greater reliability.

It's worth noting that all of our other APIs use REST, so moving to a RESTful architecture will align this API with the rest of our ecosystem. This will make it easier to maintain and enhance our APIs over time, as well as providing a more consistent experience for API customers.
```

ce-dmelnych replied:
Overview updated

marcosgvieira reviewed (changes requested):
I'm missing the field changes on the document like processDefinitionId to processDefinitionKey, creationTime to creationDate, @ce-dmelnych can you please add this changes for each section that we have changed the names, so will make it easier for the customers to understand the mappings between the fields?

ce-dmelnych commented:
> I'm missing the field changes on the document like processDefinitionId to processDefinitionKey, creationTime to creationDate, @ce-dmelnych can you please add this changes for each section that we have changed the names, so will make it easier for the customers to understand the mappings between the fields?

Yep, added notes about the renamings to `GraphQL to REST API migration` page and schema descriptions.

marcosgvieira reviewed (approved):
I'm happy with the changes, thanks for the work on that @ce-dmelnych!

akeller commented:
Looks like there was one link issue detected by the CI. If that continues to be a problem, feel free to pull me in to have a look so we can get this merged!

marcosgvieira commented:
@akeller I just had a look and I'm not sure from where the link is coming just from the error message:

`Provided URLs failing:
http://172.17.0.2:8080/docs/next/self-managed/identity/user-guide/groups/creating-authorizations-for-a-group/ (HTTP 404)`

Can you give us a hand to identify if this is the url failing or I'm just misinterpreting the error?

akeller commented:
I'm on it! I'll take care of this @marcosgvieira 😄

akeller commented:
Since we are working our way to the 8.2 docs release, I will merge this and create an issue for DevEx to review this content soon.
